web/src/compose_state.js

```diff
@@ -97,11 +97,9 @@ export function set_stream_name(stream_name) {
     compose_recipient.set_selected_recipient_id(stream_id);
 }
 
-export function set_compose_recipient_id(value) {
-    let recipient_id = compose_recipient.DIRECT_MESSAGE_ID;
-    if (typeof value === "number") {
-        // value is stream name
-        recipient_id = sub_store.maybe_get_stream_name(value) || "";
+export function set_compose_recipient_id(recipient_id) {
+    if (typeof value !== "number") {
```

Comment:
You mean `typeof recipient_id`, right?

Reply:
Oh yeah, I'm really surprised I didn't catch that with the linter or tests. Thanks!